plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/comunnication/MavenMessagesHandler.java

```diff
@@ -54,52 +42,42 @@
  */
 @Singleton
 public class MavenMessagesHandler {
-
     private final EventBus                  eventBus;
-    private final DtoFactory                factory;
     private final BackgroundLoaderPresenter dependencyResolver;
     private final PomEditorReconciler       pomEditorReconciler;
     private final ProcessesPanelPresenter   processesPanelPresenter;
-    private final CommandConsoleFactory     commandConsoleFactory;
     private final AppContext                appContext;
 
+    private DefaultOutputConsole outputConsole;
+
     @Inject
     public MavenMessagesHandler(EventBus eventBus,
-                                DtoFactory factory,
+                                MavenJsonRpcHandler mavenJsonRpcHandler,
                                 BackgroundLoaderPresenter dependencyResolver,
                                 PomEditorReconciler pomEditorReconciler,
-                                WsAgentStateController wsAgentStateController,
                                 ProcessesPanelPresenter processesPanelPresenter,
                                 CommandConsoleFactory commandConsoleFactory,
                                 AppContext appContext) {
         this.eventBus = eventBus;
-        this.factory = factory;
-
         this.dependencyResolver = dependencyResolver;
         this.pomEditorReconciler = pomEditorReconciler;
         this.processesPanelPresenter = processesPanelPresenter;
-        this.commandConsoleFactory = commandConsoleFactory;
         this.appContext = appContext;
 
-        handleOperations(wsAgentStateController);
+        mavenJsonRpcHandler.addTextHandler(this::handleTextNotification);
+        mavenJsonRpcHandler.addStartStopHandler(this::handleStartStop);
+        mavenJsonRpcHandler.addPercentHandler(this::handlePercentNotification);
+        mavenJsonRpcHandler.addProjectsUpdateHandler(this::handleUpdate);
+        mavenJsonRpcHandler.addArchetypeOutputHandler(this::onMavenArchetypeReceive);
+
+        handleOperations();
+        outputConsole = (DefaultOutputConsole)commandConsoleFactory.create("Maven Archetype");
     }
 
-    private void handleOperations(final WsAgentStateController wsAgentStateController) {
+    private void handleOperations() {
         eventBus.addHandler(WsAgentStateEvent.TYPE, new WsAgentStateHandler() {
```

Comment:
I think we can remove this handler 
 `eventBus.addHandler(WorkspaceStoppedEvent.TYPE, event -> dependencyResolver.hide());`
will be enough